lnwallet/channel.go

```diff
@@ -1328,12 +1345,44 @@ func (lc *LightningChannel) fetchCommitmentView(remoteChain bool,
 		numHTLCs++
 	}
 
+	// Initiate feePerKw to the last committed fee for this chain.
+	feePerKw := commitChain.tail().feePerKw
+
+	// Check if any fee updates have taken place since that last commitment.
+	if lc.channelState.IsInitiator {
```

Comment:
Discussed offline and came to the agreement that the model I described would require a queue of sorts. We've decided to go with this approach for now, and agreed that we can revisit the implementation given the adequate set of unit tests.